### PR TITLE
Offer an API option to display new warnings on problems with URDF to SDF conversions (sdf12)

### PR DESCRIPTION
**Pull request by j-rivero**
# 🎉 New feature

See #1171 . It was requested to implement first the feature in sdf12.

## Summary

Instead of using the overloading approach in #1171, in this branch the API is nicer using ParserConfig options:
```c++
  /// \brief set the enableNewWarnings flag to _enable value
  public: void URDFSetNewWarnings(const bool _enable);

  /// \brief get the enableNewWarnings flag value;
  public: bool URDFEnableNewWarnings() const;
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/parser_urdf.cc [hunk 1]
```diff
@@ -2655,42 +2657,58 @@ void URDF2SDF::ListSDFExtensions(const std::string &_reference)
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+void _DisplayDbgOrWarning(const std::string &_msg,
+                          const bool _enableNewWarnings) {
+  if (_enableNewWarnings)
+    sdfwarn << _msg;
+  else
+    sdfdbg << _msg;
```
Comment (scpeters) [inline src/parser_urdf.cc:2666]: this looks like an enforcement policy to me, so I think we could use that mechanism explicitly, but only the `LOG` and `WARN` options

File: include/sdf/ParserConfig.hh [hunk 2]
```diff
@@ -174,6 +174,12 @@ class SDFORMAT_VISIBLE ParserConfig
   /// \brief Get the preserveFixedJoint flag value.
   public: bool URDFPreserveFixedJoint() const;
 
+  /// \brief set the enableNewWarnings flag to _enable value
+  public: void URDFSetNewWarnings(const bool _enable);
+
+  /// \brief get the enableNewWarnings flag value;
+  public: bool URDFEnableNewWarnings() const;
```
Comment (scpeters) [inline include/sdf/ParserConfig.hh:181]:
```suggestion
  public: EnforcementPolicy URDFPolicy() const;
```

File: include/sdf/ParserConfig.hh [hunk 3]
```diff
@@ -174,6 +174,12 @@ class SDFORMAT_VISIBLE ParserConfig
   /// \brief Get the preserveFixedJoint flag value.
   public: bool URDFPreserveFixedJoint() const;
 
+  /// \brief set the enableNewWarnings flag to _enable value
+  public: void URDFSetNewWarnings(const bool _enable);
```
Comment (scpeters) [inline include/sdf/ParserConfig.hh:178]:
```suggestion
  public: void SetURDFEnforcementPolicy(EnforcementPolicy _policy);
```

File: src/parser_urdf.cc (hunk 1 above)
Comment (j-rivero) [reply src/parser_urdf.cc:2672]: Changed in [b05e7b7](https://github.com/gazebosim/sdformat/pull/1192/commits/b05e7b72b2b48277c026b7290166eaffc0cd074a)

File: include/sdf/ParserConfig.hh (hunk 2 above)
Comment (j-rivero) [reply include/sdf/ParserConfig.hh]: Changed in [b05e7b7](https://github.com/gazebosim/sdformat/pull/1192/commits/b05e7b72b2b48277c026b7290166eaffc0cd074a)

File: include/sdf/ParserConfig.hh (hunk 3 above)
Comment (j-rivero) [reply include/sdf/ParserConfig.hh]: Changed in [b05e7b7](https://github.com/gazebosim/sdformat/pull/1192/commits/b05e7b72b2b48277c026b7290166eaffc0cd074a)

**Comment by azeey** [comment]
Closing in favor of #1238

**Pull request closed without merging**